src/server/server.rs

```diff
@@ -94,19 +99,28 @@ impl MystService for TimeseriesService {
 #[tokio::main]
 async fn main() -> Result<(), Box<dyn std::error::Error>> {
     let config = Config::new();
+    let lib = libloading::Library::new(&config.plugin_path).expect("load library");
+    let new_service: libloading::Symbol<
+        fn(&str, &str, &str) -> Box<dyn MetricsReporter + Send + Sync>,
```

Comment:
nit: `MetricsReporter` already requires `Send + Sync` by definition, not sure you need them here.